src/main/java/org/opensearch/searchrelevance/ml/connector/LLMConnector.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.searchrelevance.ml.connector;
+
+import java.util.Map;
+
+/**
+ * Interface for LLM connector implementations that handle different LLM providers
+ */
+public interface LLMConnector {
```

Comment:
this is really good abstraction, thank you for allocating time and addressing this